### PR TITLE
Update stream-extensibility section docs

### PR DESCRIPTION
**Pull request by xiazen**
fix for #4434 . the section is outdated, still in the 1.5 and before state.

### CODE REVIEW

File: src/Documentation/Orleans-Streams/Streams-Extensibility.md [hunk 1]
```diff
@@ -7,40 +7,118 @@ title: Orleans Streams Extensibility
 
 There are three ways developers can extend the currently implemented behaviour of Orleans Streaming:
 
-1. Utilize or extend Stream Provider Configuration.
-2. Write a Custom Queue Adapter.
-3. Writ a New Stream Provider
+- Utilize or extend Stream Provider Configurators.
+- Write a Custom Queue Adapter.
+- Writ a New Stream Provider
```
Comment (ReubenBond) [inline src/Documentation/Orleans-Streams/Streams-Extensibility.md]: typo on write

File: src/Documentation/Orleans-Streams/Streams-Extensibility.md [hunk 2]
```diff
@@ -7,40 +7,118 @@ title: Orleans Streams Extensibility
...
+### Simple Message Stream Provider Configuration. 
+SMS Stream Provider is configured through following options, using `AddSimpleMessageStreamProvider` extension method on `ISiloHostBuilder` or `IClientBuilder`: 
 
-1. **FireAndForgetDelivery**: this option specifies if the messages sent by SMS stream producer are sent as fire and forget without the way to know if they were delivered or not. When FireAndForgetDelivery is set to false (messages are sent not as FireAndForget), the stream producer's call `stream.OnNext()` returns a Task that represents the processing status of the stream consumer. If this Task succeeds, the producer knows for sure that the message was delivered and processed successfully. If FireAndForgetDelivery is set to true, the returned Task only expresses that the Orleans runtime has accepted the message and queued it for further delivery. The default value for FireAndForgetDelivery is false.
+```csharp
+    public class SimpleMessageStreamProviderOptions
+    {
+        public bool FireAndForgetDelivery { get; set; } = DEFAULT_VALUE_FIRE_AND_FORGET_DELIVERY;
+        public const bool DEFAULT_VALUE_FIRE_AND_FORGET_DELIVERY = false;
```
Comment (ReubenBond) [inline src/Documentation/Orleans-Streams/Streams-Extensibility.md]: This code should be de-indented by one level

**Comment by xiazen** [comment]
any new comments ? if not, can we merge it ?

**Comment by ReubenBond** [comment]
Just some grammatical things, I'll push an update to this branch and merge it

**Comment by xiazen** [comment]
Thank you ! :)

**Comment by xiazen** [comment]
can we move forward with this PR? if you don't have time, you can just comment and I will go fix those grammatical errors

**Pull request merged**